tests/unit/test_utils.py

```diff
@@ -384,8 +386,8 @@ def test_rmtree_retries_for_3sec(tmpdir, monkeypatch):
     # Test a bytes object with a character that can't be decoded.
     (u'/path/déf'.encode('utf-8'), 'ascii', u"b'/path/d\\xc3\\xa9f'"),
     (u'/path/déf'.encode('utf-16'), 'utf-8',
-     u"b'\\xff\\xfe/\\x00p\\x00a\\x00t\\x00h\\x00/"
-     "\\x00d\\x00\\xe9\\x00f\\x00'"),
+     str_to_display("b{!r}".format(u'/path/déf'.encode('utf-16')))
+     if PY2 else ascii(u'/path/déf'.encode('utf-16'))),
```

Comment:
I wonder if this kind of defeats the test (you’re just comparing the results of matching implementation). Would it be possible to detect the endian, and use the correct raw bytes here?

Reply:
We can use [sys.byteorder](https://docs.python.org/3/library/sys.html#sys.byteorder) to detect the endianness

Reply:
I was myself doubtful of this approach but implemented it anyway to get some reviews. @deveshks surely, let me have a look.